### PR TITLE
Fixed colors for merchants

### PR DESCRIPTION
**Pull request by tomschw**
This PR will sort the lines and legends in the graphs in lexicographical order and the merchants now have a self-defined line color.

### CODE REVIEW

File: asset/js/app.js [hunk 1]
```diff
@@ -249,6 +249,9 @@
             getMerchantName: function(merchant_id) {
                 return ((merchant_id in merchants) && merchants[merchant_id].merchant_name) ? merchants[merchant_id].merchant_name : merchant_id.substring(0, 8);
             },
+            getMerchantColor: function(merchant_id) {
+                return (merchant_id in merchants) ? merchants[merchant_id].color : '#FFFFFF';
```
Comment (Bouncner) [inline asset/js/app.js]: Why white? That's pretty much transparent, isn't it?

File: asset/js/app.js [hunk 2]
```diff
@@ -500,34 +506,36 @@
...
+                    sortLegend(chart);
                     chart.redraw();
                 },
                 filterForID: function(chart, productID) {
                     chart.series.forEach(function(serie) {
                         dontDrawLineIfLineFiltered(chart, serie.options.id, productID);
                     });
 
-                    sortLegend(chart);
+                    // sortLegend(chart);
```
Comment (Bouncner) [inline asset/js/app.js]: ?

File: asset/js/app.js [hunk 3]
```diff
@@ -573,7 +582,27 @@
             point.merchant_id = data.value.merchant_id;
         }
 
-        function addPointToLine(chart, point, lineID, lineName, stepEnabled, lineToUse) {
+        function sortLegend(chart) {
+            var line;
+
```
Comment (Bouncner) [inline asset/js/app.js]: Too much whitespace.

File: asset/js/app.js [hunk 4]
```diff
@@ -942,7 +906,7 @@
                     title: {
                         text: 'Date'
                     },
-                    showEmpty: false
+                    showEmpty: false,
```
Comment (Bouncner) [inline asset/js/app.js]: ?

File: asset/js/app.js (hunk 1 above)
Comment (tomschw) [reply asset/js/app.js]: It is basically just a fail safe. This only happens if we have a merchant which the marketplace doesn't know, which should never happen.

Comment (tomschw) [reply asset/js/app.js]: @Bouncner Fixed issue.

**Comment by cmfcmf** [comment]
> This PR will sort the lines and legends in the graphs in lexicographical order and the merchants now have a self-defined line color.

That's so great – the constantly changing colors were so confusing!

File: asset/js/app.js [hunk 5]
```diff
@@ -250,7 +250,11 @@
                 return ((merchant_id in merchants) && merchants[merchant_id].merchant_name) ? merchants[merchant_id].merchant_name : merchant_id.substring(0, 8);
             },
             getMerchantColor: function(merchant_id) {
-                return (merchant_id in merchants) ? merchants[merchant_id].color : '#FFFFFF';
+                if (merchant_id in merchants) {
+                    return merchants[merchant_id].color;
+                }
+                console.error("Found an merchant id that is not included in our merchants from the marketplace.");
+                return '#E1E1E1';
```
Comment (Bouncner) [inline asset/js/app.js]: "included in our merchants from the marketplace"? Please rewrite.\
How did you "find" a merchant id?

Also: short comment which color this is.

Comment (tomschw) [reply asset/js/app.js]: Is it better now?

**Pull request merged**